pkg/cli/verifier/envoy_config.go

```diff
@@ -0,0 +1,248 @@
+	}
+	if t.trafficAttr.DstService != nil {
+		fmt.Fprintf(&s, "#destination service: %s ", t.trafficAttr.DstService)
+	}
+	if t.trafficAttr.DstHost != "" {
+		fmt.Fprintf(&s, "#destination host: %s ", t.trafficAttr.DstHost)
+	}
+	if t.trafficAttr.DstPort != 0 {
+		fmt.Fprintf(&s, "#destination port: %d ", t.trafficAttr.DstPort)
+	}
+	fmt.Fprintf(&s, "#destination protocol: %s ", t.trafficAttr.AppProtocol)
+
+	return s.String()
+}
+
+// EnvoyConfigVerifier implements the Verifier interface for Envoy configs
+type EnvoyConfigVerifier struct {
+	stdout     io.Writer
+	stderr     io.Writer
+	kubeClient kubernetes.Interface
+	meshConfig *configv1alpha2.MeshConfig
+	configAttr configAttribute
+}
+
+// NewEnvoyConfigVerifier returns a Verifier for Envoy config verification
+func NewEnvoyConfigVerifier(stdout io.Writer, stderr io.Writer, kubeClient kubernetes.Interface,
+	meshConfig *configv1alpha2.MeshConfig, configAttr configAttribute) Verifier {
+	return &EnvoyConfigVerifier{
+		stdout:     stdout,
+		stderr:     stderr,
+		kubeClient: kubeClient,
+		meshConfig: meshConfig,
+		configAttr: configAttr,
+	}
+}
+
+// Run executes the Envoy config verifier
+func (v *EnvoyConfigVerifier) Run() Result {
+	result := Result{
+		Context: fmt.Sprintf("Verify Envoy config for traffic [%s]", v.configAttr),
+	}
+
+	if v.configAttr.trafficAttr.SrcPod != nil {
+		// verify source
+		res := v.verifySource()
+		result.NestedResults = append(result.NestedResults, &res)
+		if res.Status != Success {
+			result.Status = Failure
+			return result
+		}
+	}
+
+	result.Status = Success
+	return result
+}
+
+func (v *EnvoyConfigVerifier) verifySource() Result {
+	result := Result{
+		Context: fmt.Sprintf("Verify Envoy config on source for traffic [%s]", v.configAttr),
+	}
+
+	config, err := v.configAttr.srcConfigGetter.Get()
+	if err != nil || config == nil {
+		result.Status = Unknown
+		result.Reason = fmt.Sprintf("Error retrieving Envoy config for pod %q", v.configAttr.trafficAttr.SrcPod)
+		return result
+	}
+
+	//
+	// Verify the config
+
+	// Check for outbound listener
+	listeners := config.Listeners.GetDynamicListeners()
+	var outboundListener xds_listener.Listener
+	for _, l := range listeners {
+		if l.Name == lds.OutboundListenerName {
+			active := l.GetActiveState()
+			if active == nil {
+				result.Status = Failure
+				result.Reason = fmt.Sprintf("Outbound listener %q on source pod %q is not active", lds.OutboundListenerName, v.configAttr.trafficAttr.SrcPod)
+				return result
+			}
+			//nolint: errcheck
+			//#nosec G104: Errors unhandled
+			active.Listener.UnmarshalTo(&outboundListener)
+			break
+		}
+	}
+
+	if outboundListener.Name != lds.OutboundListenerName {
+		result.Status = Failure
+		result.Reason = fmt.Sprintf("Outbound listener %q not found on source pod %q", lds.OutboundListenerName, v.configAttr.trafficAttr.SrcPod)
+		return result
+	}
+
+	// Next, if the destination service is known, verify it has a matching filter chain
```

Comment:
Could you share the steps to repro this? Unsure if you saw this in permissive mode or with SMI policies. 
Note that the existing checks aren't complete and are a starting point to building more verifiers.

Reply:
This was the demo without any changes or flags, so with SMI policies.

Reply:
Yes so that will result in positive verification because the tool currently does not implement the destination side verification. I am working on implementing destination side checking next, which should address this false positive.

Reply:
@nojnhuh This is correctly flagged with the destination side checks. See the [PR](https://github.com/openservicemesh/osm/pull/4672) description for an example.